server/src/main/java/org/opensearch/repositories/Repository.java

```diff
@@ -440,4 +440,17 @@ default void cloneRemoteStoreIndexShardSnapshot(
     default Map<String, Object> adaptUserMetadata(Map<String, Object> userMetadata) {
         return userMetadata;
     }
+
+    /**
+     * Checks if the repository can be reloaded inplace or not
+     * @return true if the repository can be reloaded inplace, false otherwise
+     */
+    default boolean isReloadable() {
+        return false;
+    }
+
+    /**
+     * Reload the repository inplace
+     */
+    default void reload(RepositoryMetadata repositoryMetadata, boolean compress) {}
 }
```

Comment:
Change to 
```
default void reload(RepositoryMetadata repositoryMetadata) {};
```

Reply:
Agree. I attempted to refactor this to avoid requiring `compress`. Facing some issues with `FsRepository`, which supports pulling the setting from the env in addition to the repo metadata. I will try again to find a fix for this.

Reply:
We don't need to change the behavior for just `FsRepository`. Let `compress` work as it is  .